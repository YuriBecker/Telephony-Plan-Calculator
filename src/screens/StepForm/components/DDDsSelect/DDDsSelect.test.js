import React from 'react';
import { shallow } from 'enzyme';

import DDDSelect from './index';

describe('Testing DDDSelect component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <DDDSelect
        origin={null}
        setOrigin={jest.fn()}
        destiny={null}
        setDestiny={jest.fn()}
      />
    );
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render the  Buttons', () => {
    expect(
      // wrapper.findWhere(node => node.prop('testID') === 'title')
      wrapper.find('ButtonComponent')
    ).toExist();
  });

  it('should render the title Text', () => {
    expect(
      // wrapper.findWhere(node => node.prop('testID') === 'title')
      wrapper.find('TextComponent')
    ).toExist();
  });

  it('should render the  Layout container', () => {
    expect(
      // wrapper.findWhere(node => node.prop('testID') === 'title')
      wrapper.find('LayoutComponent')
    ).toExist();
  });
});
