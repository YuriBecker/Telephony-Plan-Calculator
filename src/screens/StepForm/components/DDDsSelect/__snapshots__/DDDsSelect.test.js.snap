// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DDDSelect component should render correctly 1`] = `
<Fragment>
  <TextComponent
    category="h5"
    style={
      Object {
        "color": "#1db954",
        "marginVertical": 15,
        "textAlign": "center",
      }
    }
  >
    DDD Origem
  </TextComponent>
  <LayoutComponent
    style={
      Object {
        "backgroundColor": "#212121",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 15,
      }
    }
  >
    <ButtonComponent
      appearance="filled"
      key="011"
      onPress={[Function]}
      style={
        Object {
          "marginHorizontal": 3,
        }
      }
    >
      011
    </ButtonComponent>
    <ButtonComponent
      appearance="filled"
      key="016"
      onPress={[Function]}
      style={
        Object {
          "marginHorizontal": 3,
        }
      }
    >
      016
    </ButtonComponent>
    <ButtonComponent
      appearance="filled"
      key="017"
      onPress={[Function]}
      style={
        Object {
          "marginHorizontal": 3,
        }
      }
    >
      017
    </ButtonComponent>
    <ButtonComponent
      appearance="filled"
      key="018"
      onPress={[Function]}
      style={
        Object {
          "marginHorizontal": 3,
        }
      }
    >
      018
    </ButtonComponent>
  </LayoutComponent>
  <TextComponent
    category="h5"
    style={
      Object {
        "color": "#1db954",
        "marginVertical": 15,
        "textAlign": "center",
      }
    }
  >
    DDD Destino
  </TextComponent>
  <LayoutComponent
    style={
      Object {
        "backgroundColor": "#212121",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 15,
      }
    }
  >
    <ButtonComponent
      appearance="filled"
      disabled={true}
      key="011"
      onPress={[Function]}
      previousBtnText="Voltar"
      style={
        Object {
          "marginHorizontal": 3,
        }
      }
    >
      011
    </ButtonComponent>
    <ButtonComponent
      appearance="filled"
      disabled={true}
      key="016"
      onPress={[Function]}
      previousBtnText="Voltar"
      style={
        Object {
          "marginHorizontal": 3,
        }
      }
    >
      016
    </ButtonComponent>
    <ButtonComponent
      appearance="filled"
      disabled={true}
      key="017"
      onPress={[Function]}
      previousBtnText="Voltar"
      style={
        Object {
          "marginHorizontal": 3,
        }
      }
    >
      017
    </ButtonComponent>
    <ButtonComponent
      appearance="filled"
      disabled={true}
      key="018"
      onPress={[Function]}
      previousBtnText="Voltar"
      style={
        Object {
          "marginHorizontal": 3,
        }
      }
    >
      018
    </ButtonComponent>
  </LayoutComponent>
</Fragment>
`;
